Coding/fw/lib/Lawicel/NVMInterface.h

```diff
@@ -0,0 +1,87 @@
+/***************************************************************************************************
+  (c) NewTec GmbH 2020   -   www.newtec.de
+  $URL: https://github.com/NewTec-GmbH/esp32-can-iot $
+***************************************************************************************************/
+/**
+@addtogroup Interfaces
+@ingroup Lawicel
+@{
+@file       NVMInterface.h
+
+NVM Interface for Lawicel Protocol
+
+* @}
+***************************************************************************************************/
+#ifndef NVM_INTERFACE_H
+#define NVM_INTERFACE_H
+
+/* INCLUDES ***************************************************************************************/
+
+/* C-Interface ************************************************************************************/
+extern "C"
+{
+}
+
+/* FORWARD DECLARATIONS ***************************************************************************/
+
+class NVMInterface
+{
+public:
+  /* CONSTANTS ******************************************************************************/
+
+  /* TYPES **********************************************************************************/
+
+  /**
+  * Default constructor.
+  */
+
+  NVMInterface()
+  {
+  }
+
+  /**
+  * Default destructor
+  */
+
+  virtual ~NVMInterface()
+  {
+  }
+
+  /**
+    * Initialize Module
+    */
+  virtual uint8_t begin() = 0;
+
+  /**
+  * Terminate Module
+  */
+  virtual uint8_t end() = 0;
+
+  /**
+  * Save Data in NVM
+  */
+  virtual void saveInt(String name, int32_t value) = 0;
```

Comment:
C++ allows overloading, so you don't need different method wording like  saveX( x value), you cann call all save(x value)